pages/posts/ocaml-writer.mdx

```diff
@@ -0,0 +1,234 @@
+import Page from '@reason/pages/Posts'; export default Page;
+
+# Writer monads for cache misses in Ocaml
+
+Tracking knowledge across large areas of a codebase is a fairly common task. One example of this is keeping track of whether or not some cache miss happens when loading many files. If we model knowledge monoidally, we can use a writer monad to track this knowledge without boilerplate in a simple and threadsafe manner.
```

Comment:
I would be explicit about why a cache miss on loading files is an example of this. It's not clear to me at first glance. I think it would be clearer if you said something like "Keeping track of whether or not some cache miss happens when loading files is an example of this, in that the database layer retrieving info needs to talk to the networking layer to fetch missing info." or whatever example you had in mind